src/index.ts

```diff
@@ -0,0 +1,30 @@
+import SuperAgentClientImpl from './clients/superAgentClientImpl';
+
+export default class ClientSdkFactory {
+  /**
+   * Http client.
+   * @type {Object<SuperAgent | Axios | any>}
+   */
+  public clientImpl: any;
+
+  /**
+   * Instantiate implemented client object
+   * @param {String} client
+   * @param {Object} config
+   * @returns {Object}
+   */
+  constructor(client: string, config: any) {
+    if (!client) {
+      throw new Error('client is required');
+    }
+    
+    if (!config) {
+      throw new Error('config is required');
+    }
+
+    if (client === 'superagent') {
```

Comment:
Is this necessary? I mean does the consumer needs to know what library the SDK is using for making the requests?

Reply:
To bypass if we require.

Reply:
It means whenever we instantiate the SDK we need to provide the client name ( `new SDK('superagent', config)` )

Maybe would be better to have it as a config property like `config.client` and default that to 'superagent'.